routes/api/collectors.js

```diff
@@ -0,0 +1,74 @@
+var express = require('express');
+var router = express.Router();
+var model = require('../../models/index');
+
+function validateProduct(payload) {
+  try {
+    var errors = [];
+
+    if (Object.keys(payload).length === 0 && payload.constructor === Object) {
+      errors.push("Empty payload")
+    }
+
+    if (!payload.payload.nonce) {
+      errors.push("Payload does not have a nonce")
+    }
+
+    if (!payload.payload) {
+      errors.push("Payload does not have a payload object")
+    }
+
+    if (!payload.hostname) {
+      errors.push("Payload does not have a hostname")
+    }
+
+    if (errors.length >= 1) {
+      throw errors
+    }
+
+    return model.savePayload(payload, function(success) {
+        return success;
+      }, function(errors) {
+        return errors;
+    })
+  }
+  catch (e) {
+     return e;
+  }
+}
+
+function saveCollector(fields) {
```

Comment:
Same here, include this in the persistence (model) layer instead of routes. It is not a concern of the routing, only persistence.